### PR TITLE
translation(gh): translate exploration_rl_zh.rst

### PR DESCRIPTION
**Pull request by Khaiiiiiiiii**

### CODE REVIEW

File: source/02_algo/exploration_rl_en.rst [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+
+The following mechanisms can be used to get better exploration performance in deep reinforcement learning training when function approximation is performed by neural networks:
+
+-  Entropy
+   Loss：The agent is encouraged to perform more diverse actions by adding an additional entropy regularization term to the loss function.
+
+-  Noise-based
+   Exploration：Exploration is achieved by adding noise to the parameter space of observations, actions, and even the network.
+
+Intrinsic reward-based exploration
```
Comment (VaninaY) [inline source/02_algo/exploration_rl_en.rst:67]: capital letter for title or sub-title plz

File: source/02_algo/exploration_rl_en.rst [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+Count-Based Intrinsic Reward
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Count-based intrinsic reward adopts the simplest idea of measuring novelty by counting, i.e. each :math:`s` corresponds to a visit count\ :math:`N(s)`\  , the larger the value, the more times the agent has visited it before, that is, the exploration of:math:`s`  The more sufficient (or:math:`s` less novel). The exploration module gives an intrinsic reward that is inversely proportional to the state count.
+
+-  `Unifying Count-Based Exploration and Intrinsic
+   Motivation <https://arxiv.org/abs/1606.01868>`__
+   [1]_ uses a density model to approximate the frequency of state visits and proposes a novel pseudo-count derived from the density model algorithm.
+
+-  `#Exploration: A Study of Count-Based Exploration for Deep
```
Comment (VaninaY) [inline source/02_algo/exploration_rl_en.rst:93]: I think the "#" is extra.

File: source/02_algo/exploration_rl_en.rst [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+
+In order to alleviate the above problems, the paper [4]_ `Exploration by Random Network
+Distillation <https://arxiv.org/abs/1810.12894v1>`__\ proposes the RND algorithm, which is also an exploration method based on the prediction problem, but special is that the prediction in the RND algorithm The problem is a stochastic distillation problem related only to the observation state, not a forward or inverse dynamical model of the environment. Specifically, RND utilizes two neural networks with the same structure: a target network with fixed random initialization parameters. A predictor network network), the predictor network is used to output the predicted value of the state encoding given to the target network. Then the RND intrinsic exploration reward is defined as being proportional to the state feature predicted by the predictor network\ :math:`\hat{f}( s_t )`\ and the state feature of the target network :math:`f(s_t)`\  . For more details about RND, please refer to \ `blog <https://zhuanlan.zhihu.com/p/473676311>`__\.
```
Comment (VaninaY) [inline source/02_algo/exploration_rl_en.rst:117]: please add an space after the hypelink

File: source/02_algo/exploration_rl_en.rst [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+
+Prediction problems are often problems related to the dynamics of the environment, such as the paper [3]_ `Curiosity-driven Exploration by Self-supervised Prediction <http://proceedings.mlr.press/v70/pathak17a/pathak17a.pdf>`__ (ICM) proposed a new Intrinsic Curiosity module based on prediction error. Module, ICM), by using the inverse dynamics model and the forward dynamics model to learn a new feature space on the original problem space, so that the learned feature space only encodes the part that affects the agent's decision-making, while ignoring the environment noise and other irrelevant interference. Then on this purer feature space, the prediction error of the forward model is used to provide the intrinsic for RL training reward. For more details about ICM, please refer to \ `blog <https://zhuanlan.zhihu.com/p/473676311>`__\。
```
Comment (VaninaY) [inline source/02_algo/exploration_rl_en.rst:108]: please use English punctuation only

File: source/02_algo/exploration_rl_en.rst [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+
+NGU
+''''''''
+
+In order to solve the aforementioned problem of gradual attenuation of the exploration signal, the paper [5]_ `Never Give Up: Learning
+Directed Exploration
+Strategies <https://arxiv.org/abs/2002.06038>`__\  (Never
+Give Up)
+The agent adopts a new intrinsic reward generation mechanism that integrates the novelty of 2 dimensions: namely the life-long dimension of the inter-game intrinsic reward and the single-game dimension of the intra-game intrinsic reward , and also proposed to simultaneously learn a set of strategies with different degrees of exploration (directed
+exploratory policies) to collect more abundant samples for training. Among them, the intrinsic reward between games is maintained by maintaining an Episodic that stores the state of the game. Memory, calculated by calculating the distance between the current state and the k most similar samples in Memory. More details about NGU can be found in the blog [TODO].
```
Comment (VaninaY) [inline source/02_algo/exploration_rl_en.rst:150]: 1. please put the same content in the same link
2. please capital the content within the brackets (directed exploratory policies)

File: source/02_algo/exploration_rl_en.rst [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+
+Agent57
+''''''''
+
+Paper [6]_ `Agent57: Outperforming the Atari Human
+Benchmark <https://arxiv.org/abs/2003.13350>`__\ made the following improvements on the basis of NGU:
```
Comment (VaninaY) [inline source/02_algo/exploration_rl_en.rst:156]: In general, please add a space after the hypelink

File: source/02_algo/exploration_rl_en.rst [hunk 7]
```diff
@@ -0,0 +1,241 @@
...
+
+
+Specifically, first the state is mapped into a short discrete code (called a cell) for storage. If a new state appears or a better/shorter trajectory is found, the memory updates the corresponding state and trajectory. The agent can choose a state to return uniformly and randomly in the memory, or according to some heuristic rules, for example, it can select the returned state according to the related indicators such as the newness, the access count, the count of its neighbors in the memory and so on. Then start exploring in this state. Go-Explore repeats the above process until the task is solved, i.e. at least one successful trajectory is found.
+
+Other exploration mechanisms
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In addition to the above exploration mechanism, there are also Q-value-based exploration [10]_ and so on. Interested readers can refer to this review of exploration strategies in reinforcement learning\ `blog <https://lilianweng.github.io/posts/2020-06-07-exploration-drl/>`__\ [14]_ 。
```
Comment (VaninaY) [inline source/02_algo/exploration_rl_en.rst:178]: In general, use English punctuation please

File: source/02_algo/exploration_rl_en.rst [hunk 8]
```diff
@@ -0,0 +1,241 @@
...
+
+2. Exploration based on intrinsic reward
+
+    - Count-based intrinsic rewards
+    - Intrinsic reward based on prediction error
+    - Intrinsic rewards based on information theory
+
+3. Memory-based exploration
+
+    - Episodic Memory
```
Comment (VaninaY) [inline source/02_algo/exploration_rl_en.rst:36]: In general, please keep case uniform

File: source/02_algo/exploration_rl_en.rst [hunk 9]
```diff
@@ -0,0 +1,241 @@
...
+.. image:: images/exploration_overview.png
+   :align: center
+   :scale: 25%
+
+Classic Exploration Mechanism
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In traditional multi-armed bandit problems, commonly used classical exploration mechanisms include:
+
+-  Epsilon-Greedy：At a certain moment, all actions have a choice probability greater than 0. The agent performs random actions (exploration) with a small probability\ :math:`\epsilon`\, and performs the action with the largest Q value with a large probability\ :math:`1-\epsilon`\ (using ).
```
Comment (VaninaY) [inline source/02_algo/exploration_rl_en.rst:52]: delete the space within the brackets

File: source/02_algo/exploration_rl_en.rst [hunk 10]
```diff
@@ -0,0 +1,241 @@
...
+-  Entropy
+   Loss：The agent is encouraged to perform more diverse actions by adding an additional entropy regularization term to the loss function.
+
+-  Noise-based
+   Exploration：Exploration is achieved by adding noise to the parameter space of observations, actions, and even the network.
+
+Intrinsic reward-based exploration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+One of the more important methods in exploring mechanism design is to design special rewards to stimulate the "curiosity" of the agent. Generally, we refer to the rewards given by the environment as \**extrinsic rewards**\ , and the reward given by the exploration mechanism is called \**intrinsic reward**\.
```
Comment (VaninaY) [inline source/02_algo/exploration_rl_en.rst:70]: what‘s "**", please check the format

File: source/02_algo/exploration_rl_en.rst [hunk 11]
```diff
@@ -0,0 +1,241 @@
...
+
+We hope to achieve two things by adding this additional intrinsic reward:
+
+1.  Exploration of the state space: motivating the agent to explore more novel states. (need to evaluate the novelty of state \ :math:`s`\  )
+
+2. Exploration of the state-action space: Incentivize the agent to perform actions that help reduce uncertainty about the environment. (Need to evaluate state-action pairs:math:`(s,a)` novelty)
+
+First give the definition of \**novelty**\ qualitatively:
```
Comment (VaninaY) [inline source/02_algo/exploration_rl_en.rst:78]: In general, please check "**" format. (I think maybe you are trying to make the text bold？）

File: source/02_algo/exploration_rl_en.rst [hunk 12]
```diff
@@ -0,0 +1,241 @@
...
+Paper [6]_ `Agent57: Outperforming the Atari Human
+Benchmark <https://arxiv.org/abs/2003.13350>`__\ made the following improvements on the basis of NGU:
+
+-  Parameterization of the Q function: The Q network is divided into two parts, and the Q value corresponding to the intrinsic reward and the Q value corresponding to the extrinsic reward are learned respectively.
+
+-  NGU uses different Q functions (also called strategies) with equal probability, and uses meta-controller to adaptively select Q functions corresponding to different reward discount factors and intrinsic reward weight coefficients to balance exploration and utilization.
+
+-  Finally used a larger Backprop Through Time Window Size.
+
+Explore Directly
```
Comment (VaninaY) [inline source/02_algo/exploration_rl_en.rst:164]: maybe “Direct Exploration”

File: source/02_algo/exploration_rl_en.rst [hunk 13]
```diff
@@ -0,0 +1,241 @@
...
+However, count-based methods for measuring novelty have many obvious limitations:
+
+-  There is no simple counting method for high-dimensional continuous observation space and continuous action space.
+
+-  The visit count does not accurately measure the agent's awareness of :math:`( s,a )`.
+
+Intrinsic Reward Based on Prediction Error
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Intrinsic reward based on prediction error is  \ **to use the prediction error of a state on a prediction problem (usually a supervised learning problem) to measure novelty**\. According to the characteristics of neural network fitting datasets in supervised learning, if the agent has a larger prediction error in a certain state \ :math:`s`\, the approximate state space in the vicinity of the state \ :math:`s`\. The number of previous visits by the upper agent is small, so the state\ :math:`s`\ is more novel.
```
Comment (karroyan) [inline source/02_algo/exploration_rl_en.rst:106]: Grammar mistake in "According to the characteristics of neural network fitting datasets in supervised learning, if the agent has a larger prediction error in a certain state \ :math:`s`\, the approximate state space in the vicinity of the state \ :math:`s`\."

File: source/02_algo/exploration_rl_en.rst [hunk 14]
```diff
@@ -0,0 +1,241 @@
...
+
+In order to alleviate the above problems, the paper [4]_ `Exploration by Random Network
+Distillation <https://arxiv.org/abs/1810.12894v1>`__\  proposes the RND algorithm, which is also an exploration method based on the prediction problem, but special is that the prediction in the RND algorithm The problem is a stochastic distillation problem related only to the observation state, not a forward or inverse dynamical model of the environment. Specifically, RND utilizes two neural networks with the same structure: a target network with fixed random initialization parameters. A predictor network network), the predictor network is used to output the predicted value of the state encoding given to the target network. Then the RND intrinsic exploration reward is defined as being proportional to the state feature predicted by the predictor network\ :math:`\hat{f}( s_t )`\ and the state feature of the target network :math:`f(s_t)`\  . For more details about RND, please refer to \ `blog <https://zhuanlan.zhihu.com/p/473676311>`__\.
```
Comment (karroyan) [inline source/02_algo/exploration_rl_en.rst:117]: Grammar mistake: "but special is that the prediction in the RND algorithm" and "a target network with fixed random initialization parameters. A predictor network network)"

File: source/02_algo/exploration_rl_en.rst [hunk 15]
```diff
@@ -0,0 +1,241 @@
...
+
+Intrinsic Rewards Based on Information Theory
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To encourage exploration, another idea is to design intrinsic rewards based on information theory.
+The paper [11]_ introduced Variational information maximizing exploration (VIME), the core idea is the maximization of information gain about the agent's belief of environment dynamics, using variational inference in Bayesian neural networks, which can efficiently handle continuous state and action spaces.
+The paper [12]_ proposes the EMI algorithm (Exploration with Mutual Information), which does not learn representations through the usual encoding/decoding raw state or action space, but learns the relationship between states and actions by maximizing the mutual information between related state-action representations. sign,
```
Comment (karroyan) [inline source/02_algo/exploration_rl_en.rst:124]: Why "sign,"?

File: source/02_algo/exploration_rl_en.rst [hunk 16]
```diff
@@ -0,0 +1,241 @@
...
+They experimentally verified that the forward prediction signal extracted in such a representation space can guide exploration well.
+In addition, there are also methods such as DIYAN [13]_, which is based on the objective function of mutual information to learn skill variables, which can automatically learn the distribution of state and skill by setting intrinsic rewards related to mutual information without external rewards. In subsequent hierarchical learning, imitation learning and exploration tasks.
```
Comment (karroyan) [inline source/02_algo/exploration_rl_en.rst:126]: The last sentence is incomplete.

File: source/02_algo/exploration_rl_en.rst [hunk 17]
```diff
@@ -0,0 +1,241 @@
...
+
+Memory-Based Exploration
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Intrinsic reward-based exploration methods such as ICM and RND propose to measure the novelty of a state by predicting the error of the problem, providing a large intrinsic reward for a state with a large novelty and promoting exploration. These methods are difficult to explore under many sparse reward settings. Good results have been achieved on the task of , but there is a problem: \ **As the number of training steps of the agent increases, the prediction error of the prediction problem begins to decrease, and the exploration signal becomes smaller, that is, the agent is no longer encouraged to visit a certain Some states, but it is possible that these states are the states that must be visited to obtain extrinsic rewards**\ . And there may also be the following problems:
```
Comment (karroyan) [inline source/02_algo/exploration_rl_en.rst:131]: Doesn't match the original text: "These methods are difficult to explore under many sparse reward settings. Good results have been achieved on the task of , but there is a problem"
 Delete "are the states that" in "that these states are the states that must be visited to obtain extrinsic rewards"

File: source/02_algo/exploration_rl_en.rst [hunk 18]
```diff
@@ -0,0 +1,241 @@
...
+
+The exploration mechanism of storage-based exploration explicitly uses a Memory to maintain the historical state, and then gives the intrinsic reward value of the current state according to a certain metric between the current state and the historical state.
+
+Episodic Memory
+^^^^^^^^^^^^^^^^^
+
+NGU
+''''''''
+
+In order to solve the aforementioned problem of gradual attenuation of the exploration signal, the paper [5]_ `Never Give Up: Learning
```
Comment (karroyan) [inline source/02_algo/exploration_rl_en.rst:145]: Incomplete sentence.

File: source/02_algo/exploration_rl_en.rst [hunk 19]
```diff
@@ -0,0 +1,241 @@
...
+Directed Exploration
+Strategies <https://arxiv.org/abs/2002.06038>`__\  (Never
+Give Up)
+The agent adopts a new intrinsic reward generation mechanism that integrates the novelty of 2 dimensions: namely the life-long dimension of the inter-game intrinsic reward and the single-game dimension of the intra-game intrinsic reward , and also proposed to simultaneously learn a set of strategies with different degrees of exploration (directed
+exploratory policies) to collect more abundant samples for training. Among them, the intrinsic reward between games is maintained by maintaining an Episodic that stores the state of the game. Memory, calculated by calculating the distance between the current state and the k most similar samples in Memory. More details about NGU can be found in the blog [TODO].
```
Comment (karroyan) [inline source/02_algo/exploration_rl_en.rst:150]: Delete ". Memory"
"maintaining" and "calculating" may be unnecessary.

File: source/02_algo/exploration_rl_en.rst [hunk 20]
```diff
@@ -0,0 +1,241 @@
...
+
+Agent57
+''''''''
+
+Paper [6]_ `Agent57: Outperforming the Atari Human
+Benchmark <https://arxiv.org/abs/2003.13350>`__\ made the following improvements on the basis of NGU:
```
Comment (karroyan) [inline source/02_algo/exploration_rl_en.rst:156]: Need a space after hyperlink

File: source/02_algo/exploration_rl_en.rst [hunk 21]
```diff
@@ -0,0 +1,241 @@
...
+
+-  Finally used a larger Backprop Through Time Window Size.
+
+Direct Exploration
+^^^^^^^^^^^^^^^^^^^^
+
+Go-Explore
+''''''''''''''''
+
+`Go-Explore <https://www.nature.com/articles/s41586-020-03157-9>`__ [8]_ [9]_  pointed out that there are currently two factors hindering the agent's exploration: forgetting how to reach a previously visited state (detachment); the agent cannot first return to a certain state, and then start exploration (derailment) from that state. For this reason, the author proposes a simple mechanism of\ **remembering a state, returning to that state, and exploring**\  from that state to deal with the above problem: by maintaining a memory of the states of interest and the trajectory leading to these states, The agent can return (assuming the simulator is deterministic) to these promising states and continue stochastic exploration from there. Novelty is measured by the prediction error of a state on a prediction problem (usually a supervised learning problem)
```
Comment (karroyan) [inline source/02_algo/exploration_rl_en.rst:170]: Need a space before ** format.

File: source/02_algo/exploration_rl_en.rst [hunk 22]
```diff
@@ -0,0 +1,241 @@
...
+
+
+Specifically, first the state is mapped into a short discrete code (called a cell) for storage. If a new state appears or a better/shorter trajectory is found, the memory updates the corresponding state and trajectory. The agent can choose a state to return uniformly and randomly in the memory, or according to some heuristic rules, for example, it can select the returned state according to the related indicators such as the newness, the access count, the count of its neighbors in the memory and so on. Then start exploring in this state. Go-Explore repeats the above process until the task is solved, i.e. at least one successful trajectory is found.
+
+Other Exploration Mechanisms
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In addition to the above exploration mechanism, there are also Q-value-based exploration [10]_ and so on. Interested readers can refer to this review of exploration strategies in reinforcement learning\ `blog <https://lilianweng.github.io/posts/2020-06-07-exploration-drl/>`__\ [14]_ .
```
Comment (karroyan) [inline source/02_algo/exploration_rl_en.rst:178]: Need a space before hyperlink

**Pull request merged**